Supertext.Base.Core.Specs/Http/UriBuilderTests.cs

```diff
@@ -0,0 +1,134 @@
+﻿using FakeItEasy;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using UriBuilder = Supertext.Base.Net.Http.UriBuilder;
+
+namespace Supertext.Base.Net.Specs.Http
+{
+    [TestClass]
+    public class UriBuilderTests
+    {
+        private UriBuilder _testee;
+        private IHttpContextAccessor _contextAccessor;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            _contextAccessor = A.Fake<IHttpContextAccessor>();
+            _testee = new UriBuilder(_contextAccessor);
+        }
+
+        [TestMethod]
+        public void CreateAbsoluteUri_SimpleUrlIsGiven_Created()
+        {
+            const string relative = "api/v1/order/1234";
+            SetupHttpContext("https://www.supertext.ch");
+
+            var result = _testee.CreateAbsoluteUri(relative);
+
+            result.AbsoluteUri.Should().Be("https://www.supertext.ch/api/v1/order/1234");
+        }
+
+        [TestMethod]
+        public void CreateAbsoluteUri_UrlWithSubdomainAndApiMethodIsGiven_Created()
+        {
+            const string relative = "api/v1/order/1234";
+            SetupHttpContext("https://www.cat.supertext.ch/api/v1/order/1234");
+
+            var result = _testee.CreateAbsoluteUri(relative);
+
+            result.AbsoluteUri.Should().Be("https://www.cat.supertext.ch/api/v1/order/1234");
+        }
+
+        [TestMethod]
+        public void CreateAbsoluteUri_UrlWithQueryStringIsGiven_Created()
+        {
+            const string relative = "/api/v1/order?orderId=123123";
```

Comment:
I would prefer if the query-strings were different in these two strings. When they are the same you can't prove that it's taking the correct/expected query-string.

Reply:
Good point. Is fixed.